src/fields.jl

```diff
@@ -262,7 +317,7 @@ function ∂x!(output::Field1D{Face}, input::Field1D{Face, Grid1D{Periodic}})
     
     fill_halos!(output)
     
-    𝐼x!(output, output)
+    𝐼x!(output, input)
```

Comment:
why did you change this? what's the thinking behind it?

Reply:
This was my mistake. I didn't think about it in detail. I've corrected it in the recent PR.

Reply:
I was referring to this when I said "it's alarming that the tests didn't catch it."

Reply:
I see. Maybe it's because earlier, we didn't have tests for these cases. We only had tests for cases where the output and input were living on different grids.

Reply:
The answer is there in Github. :)
I thought I had included tests for this cases to be honest. I'll have a look.